r2i/tensorflow/engine.cc

```diff
@@ -226,5 +240,77 @@ Engine::~Engine () {
   this->Stop();
 }
 
+float *Engine::GetTensorData(TF_Operation *operation,
+                             std::shared_ptr<TF_Tensor> tensor, RuntimeError &error) {
+
+  if (nullptr == tensor) {
+    error.Set (RuntimeError::Code::NULL_PARAMETER,
+               "Invalid tensor");
+    return nullptr;
+  }
+
+  TF_Output output = { .oper = operation, .index = 0 };
+  TF_DataType type = TF_OperationOutputType(output);
+
+  if (TF_FLOAT != type) {
+    error.Set (RuntimeError::Code::INCOMPATIBLE_MODEL,
+               "The output of this model is not floating point");
+    return nullptr;
+  }
+
+  return static_cast<float *>(TF_TensorData(tensor.get()));
```

Comment:
This is a very dangerous thing to do. You are separating the pointer from it's shared_ptr, why?

Reply:
I've reverted this change.